### PR TITLE
feat: simple `IPC` test

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

IPC didn't have any tests for making a request

## Solution

Adds a simple sanity test using the geth binary (we can switch it out for a mock later)

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: .github/scripts/install_test_binaries.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            zip="$name.zip"
+            curl -so "$zip" "https://gethstore.blob.core.windows.net/builds/$zip"
+            unzip "$zip"
+            mv -f "$name/geth.exe" ./
+            rm -rf "$name" "$zip"
+            ;;
+    esac
+}
+
+main
```
Comment (DaniPopes) [inline .github/scripts/install_test_binaries.sh:53]:
```suggestion
main

```

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: .github/scripts/install_test_binaries.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    mkdir -p "$BIN_DIR"
+    cd "$BIN_DIR"
+    export PATH="$BIN_DIR:$PATH"
+    if [ "$GITHUB_PATH" ]; then
+        echo "$BIN_DIR" >> "$GITHUB_PATH"
+    fi
+
+    install_geth &
+    g=$!
+    wait $g $!
```
[the remark this answers is not in the document]
Comment (prestwich) [reply .github/scripts/install_test_binaries.sh:25]: i would really love not to install geth in our tests, is there an expedient way we can do this with a mock? 😮‍💨

**Review by Evalir — COMMENTED** [review]
@prestwich seems like the ipc client doesn't read the stream at all and it times out? https://github.com/alloy-rs/alloy/actions/runs/7119215013/job/19383813553

**Review by Evalir — COMMENTED** [review]
test passes now @prestwich — any chance we can merge like this and clean up geth later?

**Pull request merged**